sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala

```diff
@@ -105,3 +106,34 @@ case class RegrAvgY(left: Expression, right: Expression)
       newLeft: Expression, newRight: Expression): RegrAvgY =
     this.copy(left = newLeft, right = newRight)
 }
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(y, x) - Returns the number of non-null number pairs in a group, where `y` is the dependent variable and `x` is the independent variable.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(y, x) FROM VALUES (1, 2), (2, 2), (2, 3), (2, 4) AS tab(y, x);
+       0.2727272727272727
+      > SELECT _FUNC_(y, x) FROM VALUES (1, null) AS tab(y, x);
+       NULL
+      > SELECT _FUNC_(y, x) FROM VALUES (null, 1) AS tab(y, x);
+       NULL
+      > SELECT _FUNC_(y, x) FROM VALUES (1, 2), (2, null), (2, 3), (2, 4) AS tab(y, x);
+       0.7500000000000001
+      > SELECT _FUNC_(y, x) FROM VALUES (1, 2), (2, null), (null, 3), (2, 4) AS tab(y, x);
+       1.0
+  """,
+  group = "agg_funcs",
+  since = "3.3.0")
+// scalastyle:on line.size.limit
+case class RegrR2(x: Expression, y: Expression) extends PearsonCorrelation(x, y, true) {
+  override def prettyName: String = "regr_r2"
+  override val evaluateExpression: Expression = {
+    lazy val corr = ck / sqrt(xMk * yMk)
+    If(n === 0.0, Literal.create(null, DoubleType),
+      If(n === 1.0, divideByZeroEvalResult, corr * corr))
```

Comment:
`divideByZeroEvalResult` is for legacy behaviors, which we don't need to follow in new functions.

Reply:
BTW, is there any reference implementation or algorithm introduction to explain this `corr * corr`?

Reply:
![image](https://user-images.githubusercontent.com/8486025/157394904-40a917f9-aa85-43f8-8d95-6686c79fb864.png)

We know how to evaluate corr, reference https://github.com/apache/spark/blob/52e7602344036a649bb19ed642dbd74bc7ee9cb1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Corr.scala#L126